### PR TITLE
Ruby 3 upgrade

### PR DESCRIPTION
**Pull request by tomas-stefano**
## Ruby 3 compatibility

For now everything looks good on Ruby 3 and I had to fix only the specs with the error 'class variable access from toplevel'.

Added rubocop in a similar config from Apply.

Let me know your thoughts.

### CODE REVIEW

**Comment by tomas-stefano** [comment]
upgrades-gif

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+Bundler/OrderedGems:
+  Enabled: false
+
+Style/AndOr:
+  EnforcedStyle: conditionals
+
+Style/StringLiterals:
+  Enabled: true
+  EnforcedStyle: single_quotes
```
Comment (carlosmartinez) [inline .rubocop.yml:12]: sigh. our day will come.

File: .travis.yml [hunk 2]
```diff
@@ -6,6 +6,8 @@ rvm:
   - 2.4
   - 2.5
   - 2.6
+  - 2.7
+  - 3.1
```
Comment (carlosmartinez) [inline .travis.yml:10]: shall we stop using travis at some point?

**Review by carlosmartinez — APPROVED** [review]

**Review by carlosmartinez — APPROVED** [review]

File: Guardfile [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 guard 'rspec', version: 2, cli: '-c -f d' do
-  watch(%r{^spec/.+_spec\.rb$})
-  watch(%r{^lib/(.+)\.rb$})     { |m| "spec/lib/#{m[1]}_spec.rb" }
-  watch('spec/spec_helper.rb')  { 'spec' }
+  watch(/^spec\/.+_spec\.rb$/)
+  watch(/^lib\/(.+)\.rb$/) { |m| "spec/lib/#{m[1]}_spec.rb" }
```
Comment (elliotcm) [inline Guardfile:5]: Hmm.  I'm kinda pro the other style for regexes with paths in them.

File: lib/rspec/flakey_spec.rb [hunk 4]
```diff
@@ -17,7 +17,11 @@ def initialize(example, attempts, retry_count)
     end
 
     def as_json
-      { attempts: @attempts, retry_count: @retry_count, location: @example.location, messages: messages }
+      { attempts:, retry_count:, location:, messages: }
+    end
+
+    def location
+      @example.location
```
Comment (elliotcm) [inline lib/rspec/flakey_spec.rb]: Should this be private?  If not should we give it an attr-reader and set it in the initializer?

File: lib/rspec/retry.rb [hunk 5]
```diff
@@ -66,9 +65,9 @@ def run
 
       report_on_flakey_examples if retry_reporter_data.any? && attempts < retry_count
     end
-    # rubocop:enable Metrics/AbcSize, Metrics/CyclomaticComplexity
+  # rubocop:enable Metrics/AbcSize, Metrics/CyclomaticComplexity
```
Comment (elliotcm) [inline lib/rspec/retry.rb]: If we're not disabling this anymore we probs don't have to re-enable it?

File: spec/lib/rspec/retry_spec.rb [hunk 6]
```diff
@@ -25,9 +38,11 @@ def count_up
     @count += 1
   end
 
+  # rubocop:disable Naming/AccessorMethodName
```
Comment (elliotcm) [inline spec/lib/rspec/retry_spec.rb]: IMO this is a bit cleaner if you put it inline on line 42, then you don't have to re-enable it.

File: lib/rspec/retry/version.rb [hunk 7]
```diff
@@ -2,6 +2,6 @@
 
 module RSpec
   class Retry
-    VERSION = '0.6.2'
+    VERSION = '0.7.0'
```
Comment (elliotcm) [inline lib/rspec/retry/version.rb:5]: What behaviour has been added to warrant a minor bump?

**Review by elliotcm — CHANGES_REQUESTED** [review]

File: lib/rspec/retry/version.rb (hunk 7 above)
Comment (tomas-stefano) [reply lib/rspec/retry/version.rb:5]: I considered the ruby upgrade to be a minor bump instead of any bugfix. Wdyt?

Comment (elliotcm) [reply lib/rspec/retry/version.rb:5]: I guess it lets you use ruby 3 stuff.

**Review by elliotcm — APPROVED** [review]

**Pull request merged**